lib/ansible/config/base.yml

```diff
@@ -685,6 +685,61 @@ DEFAULT_INTERNAL_POLL_INTERVAL:
       Higher values are more suitable for Ansible usage in automation scenarios,
       when UI responsiveness is not required but CPU usage might be a concern.
     - "The default corresponds to the value hardcoded in Ansible <= 2.1"
+DEFAULT_INTERPRETER_PYTHON:
+  name: Python interpreter path (or automatic discovery behavior) used for module execution
+  default: auto_legacy
+  env: [{mame: ANSIBLE_PYTHON_INTERPRETER}]
+  ini:
+  - {key: interpreter_python, section: defaults}
+  yaml:
+  - {key: defaults.interpreter_python}
+  vars:
+  - {name: ansible_python_interpreter}
+  version_added: "2.8"
+  description:
+  - Path to the Python interpreter to be used for module execution on remote targets, or an automatic discovery mode.
+    Supported discovery modes are ``auto``, ``auto_silent``, and ``auto_legacy`` (the default). All discovery modes
+    employ a lookup table to use the included system Python (on distributions known to include one), falling back to a
+    fixed ordered list of well-known Python interpreter locations if a platform-specific default is not available. The
+    fallback behavior will issue a warning that the interpreter should be set explicitly (since interpreters installed
+    later may change which one is used). This warning behavior can be disabled by setting ``auto_silent``. The default
+    value of ``auto_legacy`` provides all the same behavior, but for backwards-compatibility with older Ansible releases
+    that always defaulted to ``/usr/bin/python``, will use that interpreter if present (and issue a warning that the
+    default behavior will change to that of ``auto`` in a future Ansible release.
+DEFAULT_INTERPRETER_PYTHON_DISTRO_MAP:
+  name: Mapping of known included platform pythons for various Linux distros
+  default:
+    centos: &rhelish
+      '6': /usr/bin/python
+      '8': /usr/libexec/platform-python
+    debian:
+      '7': /usr/bin/python
+    fedora:
+      '23': /usr/bin/python3
+    redhat: *rhelish
+    rhel: *rhelish
+    ubuntu:
+      '14': /usr/bin/python
+      '16': /usr/bin/python3
```

Comment:
Is there a way for users to specify dicts in their config files?  Or is this a config setting that they won't be able to override because they can't match the data format?

Reply:
There is no `ini` or `env` here, so it wouldn't be modifiable via config. Also, until we support YAML configs, it wouldn't be settable by a user even if it was exposed.